helion/_compiler/inductor_lowering.py

```diff
@@ -172,29 +200,57 @@ def convert_arg(arg: Node) -> TensorBox:
                 new_node.kwargs = {**new_node.kwargs, "_extra_args": [*nodes]}
         else:
             new_node = create_extra_node(node, buffer, [*node._input_nodes, *nodes])
+
+        # Store output index if this buffer corresponds to an output
+        if buffer.get_name() in buffer_name_to_output_index:
+            new_node.meta["output_index"] = buffer_name_to_output_index[
+                buffer.get_name()
+            ]
+
         lowering_cls = (
             PointwiseLowering
             if isinstance(buffer.data, Pointwise)
             else ReductionLowering
         )
         buffer.freeze_layout()
+
+        input_nodes: list[torch.fx.Node] = []
+        for n in [*node._input_nodes, *nodes]:
+            if n not in input_nodes:
+                input_nodes.append(n)
```

Comment:
How are we sure this will match the names below?  Where are the duplicates coming from?

Reply:
Updated below